src/ctap/data_formats.rs

```diff
@@ -31,16 +31,18 @@ impl TryFrom<cbor::Value> for PublicKeyCredentialRpEntity {
     type Error = Ctap2StatusCode;
 
     fn try_from(cbor_value: cbor::Value) -> Result<Self, Ctap2StatusCode> {
-        let mut rp_map = extract_map(cbor_value)?;
-        let rp_id = extract_text_string(ok_or_missing(rp_map.remove(&cbor_text!("id")))?)?;
-        let rp_name = rp_map
-            .remove(&cbor_text!("name"))
-            .map(extract_text_string)
-            .transpose()?;
-        let rp_icon = rp_map
-            .remove(&cbor_text!("icon"))
-            .map(extract_text_string)
-            .transpose()?;
+        destructure_cbor_map! {
+            let {
+                "id" => rp_id,
+                "icon" => rp_icon,
+                "name" => rp_name,
+            } = extract_map(cbor_value)?;
+        };
```

Comment:
I'm surprised we need a semi-colon here. Do you get an error/warning/clippy if you remove it?

Reply:
I didn't check. But to me it was more in the spirit of the `let {foo} = bar;` syntax to have a full statement, followed by a semi-colon, inside of the macro.

Reply:
Ok, it's not important. It's just that I recently got a warning (or error) that macro calls are either $foo(...); or $foo { ... }. But maybe it actually depends on whether the macro result is an expression or a toplevel item. I'm not sure.

Reply:
I got time to play a bit in the playground. Apparently what I said is only enforced for items, not for expressions. However, it's essentially similar to if and match expressions. In Rust, braces are not followed by semi-colons (struct, enum, if, match, etc.) but for expressions (if, match, etc.) it doesn't hurt to add it. I don't mind, I was just surprised to see a semi-colon there.

Reply:
Oh, I thought you meant the semi-colon inside of the macro. The one outside of the macro is indeed not necessary with this new syntax.

Reply:
Yes, the one outside which is the one relevant to the compiler and clippy (the one inside is up to us and I think we should have it to be coherent with the let syntax). If we have a clippy test and it doesn't complain I guess it's fine (and I hope it should be the same result if we have a semi-colon after an if/match otherwise I would consider it a clippy bug).

Reply:
> If we have a clippy test and it doesn't complain [...]

We currently don't have a Clippy test. IIRC this was due to some dependencies somehow triggering plenty of lints.

Reply:
It's due to Tock and clippy doesn't allow whitelisting.
But maybe the situation improved and we can add clippy. I'll have a look at it.